tests/codegen/fcode/scripts/generic_methods.py

```diff
@@ -15,21 +15,3 @@ def __del__(self):
     def translate(self, a, b):
         self.x = self.x + a
         self.y = self.y + b
-
-p = Point(0.0, 0.0)
-x=p.x
-p.x=x
-a = p.x
-a = p.x - 2
-a = 2 * p.x - 2
-a = 2 * (p.x + 6) - 2
-
-p.y = a + 5
-p.y = p.x + 5
-
-p.translate(1.0, 2.0)
-
-print(p.x, p.y)
-print(a)
-
-del p
```

Comment:
What error do you get in this section? Maybe we should keep a copy with the bug in the known_bugs folder ?

Reply:
I thing there were two errors in this file, only one is related to interfaces, I am not sure I going to test it

Reply:
That is very possible. In which case we should have 2 copies of this file. One (the one you have now) to test interfaces in functions and one (the old one, still in the errors folder) to show the other problem

Reply:
Sorry where should I keep the one with the error that I fixed, do you mean I should mark it as xfail?

Reply:
In the master branch we have the file `tests/errors/known_bugs/generic_methods.py`. It is in the errors folder so this tells us that it contains something that doesn't work but did at some point.
In this branch 1 of the 2 problems in that file are fixed. It therefore makes sense for you to create the new file `tests/codegen/fcode/scripts/generic_methods.py` so this doesn't get broken again. However if you completely remove the code for the second problem then we lose the fact that that bit should work.

So you should have 2 versions of the code.
1. The file `tests/errors/known_bugs/generic_methods.py` from the master branch in the same location which will illustrate the second problem
2. The file `tests/codegen/fcode/scripts/generic_methods.py` from this branch which shows that the first problem has been fixed

Reply:
Yes, I will create the second file, for the first one is a newly introduced bug, when a class has a generic method, I will try to fix it.

Reply:
Actually I already created a file tests/codegen/fcode/scripts/generic_methods.py, that shows the fixed problem.
The second problem occurs when we call the method.

Reply:
You didn't create a file you moved a file and then removed some information. Creating a file to show the fixed problem is good. I just don't want to lose the original file that shows the method calling problem

Reply:
Yes I see, I will try to fix the problem and then add file as it was in tests/codegen

Reply:
If you can fix the problem then great. If not then just keep this and put the original back in errors